### PR TITLE
feat: implement slotted-label-mixin

### LINKED ISSUE #39: Implement slotted-label-mixin
In certain components, namely `vaadin-checkbox` and `vaadin-radio-button` we have [this logic](https://github.com/vaadin/vaadin-checkbox/blob/7572a685d5c21ccc31322d29a048ed17e154fe94/src/vaadin-checkbox.html#L197-L211):

```js
ready() {
  super.ready();

  this.shadowRoot.querySelector('[part~="label"]').querySelector('slot')
    .addEventListener('slotchange', this._updateLabelAttribute.bind(this));

  this._updateLabelAttribute();
}

_updateLabelAttribute() {
  const label = this.shadowRoot.querySelector('[part~="label"]');
  const assignedNodes = label.firstElementChild.assignedNodes();
  if (this._isAssignedNodesEmpty(assignedNodes)) {
    label.setAttribute('empty', '');
  } else {
    label.removeAttribute('empty');
  }
}
```

We should extract it to a mixin, e.g. `@vaadin/slotted-label-mixin` and use that.

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #39

### CODE REVIEW

File: packages/slotted-label-mixin/slotted-label-mixin.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { Constructor } from '@vaadin/mixin-utils';
+
+const isEmpty = (nodes: Node[]) => {
+  return (
+    nodes.length === 0 ||
+    (nodes.length === 1 &&
+      nodes[0].nodeType === Node.TEXT_NODE &&
+      (nodes[0].textContent || '').trim() === '')
+  );
+};
```
Comment (tomivirkki) [inline packages/slotted-label-mixin/slotted-label-mixin.ts]:
```suggestion
const isEmpty = (nodes: Node[]) =>
  nodes.every(node => node.nodeType === Node.TEXT_NODE && !node.textContent?.trim());
```

File: packages/slotted-label-mixin/slotted-label-mixin.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        }
+      }
+    }
+
+    private _updateLabelPart(label: Element, slot: HTMLSlotElement) {
+      const part = label.getAttribute('part');
+      const newPart = this._defaultPart + (isEmpty(slot.assignedNodes()) ? ' label-empty' : '');
+
+      if (part !== newPart) {
+        label.setAttribute('part', newPart);
```
Comment (tomivirkki) [inline packages/slotted-label-mixin/slotted-label-mixin.ts]: Shouldn't the `part` attributes always be static? Emptyness seems more like a _state_ so something like `<div part="label" empty>` might be more logical than `<div part="label label-empty">`.

Comment (web-padawan) [reply packages/slotted-label-mixin/slotted-label-mixin.ts]: Yes that's how it previously was, but in that case users would not be able to use `::part()` selector and we don't have `:state` yet.

I'nm ok to revert it for now.

Comment (web-padawan) [reply packages/slotted-label-mixin/slotted-label-mixin.ts]: Discussed in person and decided to use `label-empty` host attribute instead.

File: packages/slotted-label-mixin/slotted-label-mixin.ts (hunk 1 above)
Comment (web-padawan) [reply packages/slotted-label-mixin/slotted-label-mixin.ts]: Thanks, changed.

File: packages/slotted-label-mixin/slotted-label-mixin.ts (hunk 2 above)
Comment (web-padawan) [reply packages/slotted-label-mixin/slotted-label-mixin.ts]: Changed.

File: packages/slotted-label-mixin/slotted-label-mixin.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+const isEmpty = (nodes: Node[]) =>
+  nodes.every(node => node.nodeType === Node.TEXT_NODE && !node.textContent?.trim());
+
+export const SlottedLabelMixin = <T extends Constructor<LitElement>>(base: T): T => {
+  class SlottedLabel extends base {
+    protected firstUpdated(props: PropertyValues) {
+      super.firstUpdated(props);
+
+      const label = this.renderRoot.querySelector('[part~="label"]');
```
Comment (tomivirkki) [inline packages/slotted-label-mixin/slotted-label-mixin.ts]: I guess this variable is no longer needed? You could query the slot element directly with `this.renderRoot.querySelector('[part~="label"] slot');`

Comment (web-padawan) [reply packages/slotted-label-mixin/slotted-label-mixin.ts]: Thanks, done.

**Pull request merged**